### PR TITLE
Add required methods in native store classes

### PR DESCRIPTION
**Pull request by Ashok12011234**
- Refactored error throws to error logs
- Refactored variable and method names
- Add function to write node indexes on node index db
- Add functions which are required for streaming algorithms

### CODE REVIEW

File: src/nativestore/NodeManager.cpp [hunk 1]
```diff
@@ -104,21 +108,32 @@ NodeManager::NodeManager(GraphConfig gConfig) {
                                  ", NodeBlock::BLOCK_SIZE: " + std::to_string(NodeBlock::BLOCK_SIZE));
         std::string errorMessage = "Node DB size does not comply to node block size Path = " + nodesDBPath;
         node_manager_logger.error(errorMessage);
-        throw std::runtime_error(errorMessage);
+    }
+    if (dbSize(relationsDBPath) % RelationBlock::BLOCK_SIZE != 0) {
+        node_manager_logger.warn("RelationsDB size: " + std::to_string(dbSize(relationsDBPath)) + ", RelationBlock::BLOCK_SIZE: " + std::to_string(RelationBlock::BLOCK_SIZE));
+        std::string errorMessage =
```
Comment (miyurud) [inline src/nativestore/NodeManager.cpp:114]: Shall we make the two lines single one? I do not see any reason for assigning the error message to a string variable first.

```
node_manager_logger.error("RelationsDB size does not comply to node block size Path = " + relationsDBPath);
```

File: src/nativestore/NodeManager.cpp [hunk 2]
```diff
@@ -104,21 +108,32 @@ NodeManager::NodeManager(GraphConfig gConfig) {
...
+                "RelationsDB size does not comply to node block size Path = " + relationsDBPath;
+        node_manager_logger.error(errorMessage);
+    }
+    if (dbSize(centralRelationsDBPath) % RelationBlock::BLOCK_SIZE != 0) {
+        node_manager_logger.warn("CentralRelationsDB size: " + std::to_string(dbSize(centralRelationsDBPath)) + ", RelationBlock::BLOCK_SIZE: " + std::to_string(RelationBlock::BLOCK_SIZE));
+        std::string errorMessage =
```
Comment (miyurud) [inline src/nativestore/NodeManager.cpp:120]: Shall we move these two lines to single line as follows?

```
node_manager_logger.error("CentralRelationsDB size does not comply to node block size Path = " + centralRelationsDBPath);
```

**Review by miyurud — CHANGES_REQUESTED** [review]
Please address all the comments I have put and reply to each one of them how you addressed those.

File: src/nativestore/NodeManager.cpp (hunk 2 above)
Comment (Ashok12011234) [reply src/nativestore/NodeManager.cpp]: Addresses with suggested approach

File: src/nativestore/NodeManager.cpp (hunk 1 above)
Comment (Ashok12011234) [reply src/nativestore/NodeManager.cpp]: Addresses with suggested approach

**Comment by Ashok12011234** [comment]
> Please address all the comments I have put and reply to each one of them how you addressed those.

done

File: src/nativestore/NodeManager.cpp (hunk 1 above)
Comment (miyurud) [reply src/nativestore/NodeManager.cpp]: Ok

**Comment by miyurud** [comment]
@Ashok12011234 Please fix the style issue

**Review by miyurud — APPROVED** [review]

**Pull request merged**